### PR TITLE
[IMP] mail: introduce emoji command in the powerbox

### PR DESCRIPTION
**Pull request by phenix-factory**
Task-2987816

Enterprise: https://github.com/odoo/enterprise/pull/32252

### CODE REVIEW

File: addons/mail/__manifest__.py [hunk 1]
```diff
@@ -59,7 +59,7 @@
 (technically: Server Actions) to be triggered for each incoming mail.
     """,
     'website': 'https://www.odoo.com/app/discuss',
-    'depends': ['base', 'base_setup', 'bus', 'web_tour'],
+    'depends': ['base', 'base_setup', 'bus', 'web_tour', 'web_editor'],
```
Comment (seb-odoo) [inline addons/mail/__manifest__.py:62]: to double check if this dependency is acceptable

File: addons/mail/static/src/web_editor_emoji_command/web_editor_emoji_command.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/** @odoo-module */
+
+import { _t } from 'web.core';
+import Wysiwyg from 'web_editor.wysiwyg';
+import { getRangePosition } from '@web_editor/js/editor/odoo-editor/src/utils/utils';
+
+Wysiwyg.include({
+    async willStart() {
+        await this._super();
+        this.messaging = await owl.Component.env.services.messaging.get();
```
Comment (seb-odoo) [inline addons/mail/static/src/web_editor_emoji_command/web_editor_emoji_command.js:10]: This `await` will delay the loading of all editors until messaging is initialized, I don't think it is desirable.

Code should be done in a non-blocking `then` ideally.

File: addons/mail/static/src/web_editor_emoji_command/web_editor_emoji_command.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            fontawesome: 'fa-smile-o',
+            priority: 1, // This is the only command in its category anyway.
+            callback: () => {
+                const position = getRangePosition(this.el, document);
+                const anchor = document.createElement("div");
+                anchor.setAttribute('class', 'web-editor-emoji-picker');
+                anchor.setAttribute('style', `position: absolute; top:${position.top}px; left: ${position.left}px`);
+                document.body.append(anchor);
+                setTimeout(() => {
+                    this.messaging.update({ emojiPicker: {} });
```
Comment (seb-odoo) [inline addons/mail/static/src/web_editor_emoji_command/web_editor_emoji_command.js:52]: async code should check for existence of records

File: addons/mail/static/src/web_editor_emoji_command/web_editor_emoji_command.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+     */
+    _getPowerboxOptions: function () {
+        const options = this._super();
+        if (!this.messaging || !this.messaging.messagingBus) {
+            return options;
+        }
+        options.categories.push({
+            name: _t('Mail'),
+            priority: 300,
+        });
```
Comment (alexkuhn) [inline addons/mail/static/src/web_editor_emoji_command/web_editor_emoji_command.js:40]: - Would use category `Messaging` (none of the already defined categories are appropriate)
- priority should be relatively high in `knowledge`, something as close as the `Knowledge` and `Media`, which are at 10 and 50 respectively.

File: addons/mail/__manifest__.py (hunk 1 above)
Comment (alexkuhn) [inline addons/mail/__manifest__.py:62]: Pretty sure we don't want mutual dependencies between `mail` or `web_editor`.
Ideally bridge module, but if not then I would much prefer to just define the `/emoji` command in the module `knowledge`, as it's essentially useful in this module. It's not a big deal that it's not available without knowledge, at least for `16.0` (we could define this bridge module in `master`)

File: addons/mail/static/src/web_editor_emoji_command/web_editor_emoji_command.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        this.odooEditor.execCommand('insert', ev.detail.emoji);
+    },
+    /**
+     * @override
+     */
+    _getPowerboxOptions: function () {
+        const options = this._super();
+        if (!this.messaging || !this.messaging.messagingBus) {
+            return options;
+        }
```
Comment (alexkuhn) [inline addons/mail/static/src/web_editor_emoji_command/web_editor_emoji_command.js:36]: No need to guard `messagingBus` (always present when `messaging` is defined)

File: addons/mail/static/src/models/messaging.js [hunk 6]
```diff
@@ -93,6 +93,9 @@ addRecordMethods('Messaging', {
         const { message, ...options } = params;
         return this.env.services.notification.add(message, options);
     },
+    onClickEmoji(ev) {
+        this.messaging.messagingBus.trigger('add_emoji', { emoji: ev.currentTarget.dataset.codepoints });
+    },
```
Comment (alexkuhn) [inline addons/mail/static/src/models/messaging.js:98]: Should maybe guard `this.messaging`, any handler might be called on non-existing record

File: addons/mail/__manifest__.py (hunk 1 above)
Comment (phenix-factory) [reply addons/mail/__manifest__.py]: I moved the emoji code to knowledge in https://github.com/odoo/enterprise/pull/32252

**Pull request closed without merging**